pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -112,12 +112,22 @@ type ArrayOrString struct {
 
 // UnmarshalJSON implements the json.Unmarshaller interface.
 func (arrayOrString *ArrayOrString) UnmarshalJSON(value []byte) error {
-	if value[0] == '"' {
+	switch value[0] {
+	case '[':
+		arrayOrString.Type = ParamTypeArray
+		if err := json.Unmarshal(value, &arrayOrString.ArrayVal); err == nil {
+			return nil
+		}
+	case '"':
 		arrayOrString.Type = ParamTypeString
-		return json.Unmarshal(value, &arrayOrString.StringVal)
+		if err := json.Unmarshal(value, &arrayOrString.StringVal); err == nil {
+			return nil
+		}
 	}
```

Comment:
For error and other cases, I use this code as default handling, use a default case will be duplicated 

```
	arrayOrString.Type = ParamTypeString
	arrayOrString.StringVal = string(value)
	arrayOrString.ArrayVal = nil
```

Reply:
I think so, for the previous code we will do `string(fileContents)` for result data, assume user write an integer into the file, it will be read as string in the end.

If I want to use ArrayorString, since we don't support int, it will through error. So I suggest for any other type we don't support we just convert them into string (it will be the same as before).

Reply:
I see you explained in #4820. It makes sense to me. You can mark this as resolved.